roles/translator/src/lib/mod.rs

```diff
@@ -1,7 +1,204 @@
+use async_channel::{bounded, unbounded};
+use futures::FutureExt;
+pub use roles_logic_sv2::{
+    mining_sv2::{ExtendedExtranonce, NewExtendedMiningJob, SetNewPrevHash, SubmitSharesExtended},
+    utils::Mutex,
+};
+use status::Status;
+use std::{
+    net::{IpAddr, SocketAddr},
+    str::FromStr,
+    sync::Arc,
+};
+
+use tokio::{sync::broadcast, task};
+use tracing::{debug, error, info};
+pub use v1::server_to_client;
+
+use proxy_config::ProxyConfig;
+
 pub mod downstream_sv1;
 pub mod error;
 pub mod proxy;
 pub mod proxy_config;
 pub mod status;
 pub mod upstream_sv2;
 pub mod utils;
+
+#[derive(Clone, Debug)]
+pub struct TranslatorSv2 {
+    upstream: async_std::net::TcpStream,
+}
+
+impl TranslatorSv2 {
+    pub fn new(upstream: async_std::net::TcpStream) -> Self {
+        Self { upstream }
+    }
+
+    pub async fn start(
+        &self,
+        config: ProxyConfig,
+    ) {
+        let (tx_status, rx_status) = unbounded();
+        // `tx_sv1_bridge` sender is used by `Downstream` to send a `DownstreamMessages` message to
+        // `Bridge` via the `rx_sv1_downstream` receiver
+        // (Sender<downstream_sv1::DownstreamMessages>, Receiver<downstream_sv1::DownstreamMessages>)
+        let (tx_sv1_bridge, rx_sv1_downstream) = unbounded::<downstream_sv1::DownstreamMessages>();
+
+        // Sender/Receiver to send a SV2 `SubmitSharesExtended` from the `Bridge` to the `Upstream`
+        // (Sender<SubmitSharesExtended<'static>>, Receiver<SubmitSharesExtended<'static>>)
+        let (tx_sv2_submit_shares_ext, rx_sv2_submit_shares_ext) =
+            bounded::<SubmitSharesExtended<'static>>(10);
```

Comment:
How are the bounded values determined?

Reply:
we can't. Maybe better to use unbounded channels.

Reply:
maybe this could be configurable in the future for tuning? mainly was curious if these were derived from some calculation

Reply:
https://github.com/stratum-mining/stratum/pull/1066#discussion_r1729539914